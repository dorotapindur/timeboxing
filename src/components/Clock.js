import React from "react";

function Clock({className = "", hours = 0, minutes = 0, seconds = 0, over = false}) {
    if (hours < 0) {hours = 0}
    if (hours > 23) {hours = 23}
    if (minutes < 0) {minutes = 0}
    if (minutes > 59) {minutes = 59}
    if (seconds < 0) {seconds = 0}
    if (seconds > 59) {seconds = 59}
    let hoursFormatted = hours.toString().padStart(2, '0');
    let minutesFormatted = minutes.toString().padStart(2, '0');
    let secondsFormatted = seconds.toString().padStart(2, '0');
    return (
        <div>
            <p className={over ? "Clock timeisup " + className : "Clock " + className}>Pozosta≈Ço {hoursFormatted}:{minutesFormatted}:{secondsFormatted}</p>
        </div>
        )
    };

export default Clock;