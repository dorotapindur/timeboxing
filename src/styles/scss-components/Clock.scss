.Clock {
    color: $main-color;
    font-weight: bold;
    margin: 0 auto 20px;
    width: max-content;
}