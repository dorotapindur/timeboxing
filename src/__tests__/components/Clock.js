import React from "react";
import Clock from "../../components/Clock";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

let root;
let clockRenderer;
describe('<Clock />', () => {
    describe('when given hours, minutes and seconds (DOM) ', () => {
        beforeEach(() => {
            root = document.createElement("div");
            ReactDOM.render(
                <Clock hours={1} minutes={10} seconds={22} />, root
            );
        });
        it('renders properly ', () => {
            expect(root.childNodes[0].nodeName).toEqual("DIV");
            expect(root.childNodes[0].firstChild.nodeName).toEqual("P");
            expect(root.childNodes[0].firstChild.className).toMatch(/Clock/);
            expect(root.childNodes[0].firstChild.textContent).toMatch(/Pozostało/);
        });
        it('renders a div element with first child p ', () => {
            expect(root.childNodes[0].nodeName).toEqual("DIV");
            expect(root.childNodes[0].firstChild.nodeName).toEqual("P");
        });
        it('sets a Clock className for p ', () => {
            expect(root.childNodes[0].firstChild.className).toMatch(/Clock/);
        });
        it('renders text of p properly ', () => {
            expect(root.childNodes[0].firstChild.textContent).toMatch(/Pozostało/);
        });
    });
    describe('when given hours, minutes and seconds (TestRenderer) ', () => {
        beforeEach(() => {
            clockRenderer = renderer.create(
                <Clock hours={1} minutes={10} seconds={22} />
            );
        });
        it('renders properly ', () => {
            console.log(clockRenderer.toJSON());
            expect(clockRenderer.toJSON()).toMatchSnapshot();
        });
        it('renders a div', () => {
            expect(clockRenderer.toJSON().type).toEqual('div');
        });
        it('sets a Clock className for p ', () => {
            expect(clockRenderer.toJSON().children[0].props).toMatchObject({"className": expect.stringMatching(/Clock/)});
        });
        it('renders text of p properly ', () => {
            expect(clockRenderer.toJSON().children[0].children).toEqual(expect.arrayContaining(["Pozostało "]));
        });
    });
});