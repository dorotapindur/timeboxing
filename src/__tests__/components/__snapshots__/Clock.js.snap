// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clock /> when given hours, minutes and seconds (TestRenderer)  renders properly  1`] = `
<div>
  <p
    className="Clock"
  >
    Pozosta≈Ço 
    <span
      className="Clock__hours"
    >
      01
    </span>
    <span
      className="Clock__colon"
    >
      :
    </span>
    <span
      className="Clock__minutes"
    >
      10
    </span>
    <span
      className="Clock__colon"
    >
      :
    </span>
    <span
      className="Clock__seconds"
    >
      22
    </span>
  </p>
</div>
`;
